relay-ourlogs/src/ourlog.rs

```diff
@@ -61,27 +93,152 @@ pub fn otel_to_sentry_log(otel_log: OtelLog) -> OurLog {
+        17..=20 => OurLogLevel::Error,
+        21..=24 => OurLogLevel::Fatal,
+        _ => OurLogLevel::from_str(&severity_text).unwrap_or(OurLogLevel::Info),
+    };
+
     OurLog {
-        timestamp_nanos: Annotated::new(otel_log.time_unix_nano),
-        observed_timestamp_nanos: Annotated::new(observed_time_unix_nano),
-        trace_id: TraceId(trace_id).into(),
-        span_id: Annotated::new(SpanId(span_id)),
+        timestamp: Annotated::new(Timestamp(timestamp)),
+        trace_id: Annotated::new(trace_id),
+        span_id: Annotated::new(span_id),
+        level: Annotated::new(level),
+        attributes: Annotated::new(attribute_data),
+        body: Annotated::new(body),
+        other: Object::default(),
         trace_flags: Annotated::new(0),
         severity_text: severity_text.into(),
         severity_number: Annotated::new(severity_number as i64),
-        attributes: attribute_data.into(),
-        body: Annotated::new(body),
-        ..Default::default()
+        timestamp_nanos: Annotated::new(otel_log.time_unix_nano),
+        observed_timestamp_nanos: Annotated::new(observed_time_unix_nano),
+    }
+}
+
+/// This fills attributes with OTel specific fields to be compatible with the otel schema.
+///
+/// This also currently backfills data into deprecated fields on the OurLog protocol in order to continue working with the snuba consumers.
+///
+/// This will need to transform all fields into attributes to be ported to using the generic trace items consumers once they're done.
+#[allow(deprecated)]
+pub fn ourlog_merge_otel(ourlog: Annotated<OurLog>) -> Annotated<OurLog> {
+    let mut ourlog = ourlog;
+    if let Some(ourlog_value) = ourlog.value_mut() {
+        if let Some(attributes) = ourlog_value.attributes.value_mut() {
+            attributes.insert(
+                "sentry.severity_number".to_string(),
+                Annotated::new(AttributeValue::IntValue(level_to_otel_severity_number(
+                    ourlog_value.level.value().cloned(),
+                ))),
+            );
+            attributes.insert(
+                "sentry.severity_text".to_string(),
+                Annotated::new(AttributeValue::StringValue(
+                    ourlog_value
+                        .level
+                        .value()
+                        .map(|level| level.to_string())
+                        .unwrap_or_else(|| "info".to_string()),
+                )),
+            );
+        }
+
+        if let Some(attributes) = ourlog_value.attributes.value() {
+            if ourlog_value.severity_text.value().is_none() {
+                if let Some(attr) = attributes.get("sentry.severity_text") {
+                    if let Some(val) = attr.value() {
+                        if let Some(str_val) = val.string_value() {
+                            ourlog_value.severity_text = Annotated::new(str_val.clone());
+                        }
+                    }
+                }
+            }
```

Comment:
@Dav1dde what's the preferred way of flattening these option chains? `and_then`?

Reply:
Yes unfortunately that's very verbose, `and_then` is usually the way if it is too big, you can make a helper function which returns `Option<T>` and then use the `?` for example:

```rs
fn foo(t: Option<Option<Option<i32>>>) -> Option<i32> {
    Some(t???)
}
```

Reply:
Thanks, I'll take a look